src/reducers/tabs.js

```diff
@@ -0,0 +1,160 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */
+
+// @flow
+
+/**
+ * Tabs reducer
+ * @module reducers/tabs
+ */
+
+import { createSelector } from "reselect";
+import move from "lodash-move";
+
+import { prefs } from "../utils/prefs";
+import {
+  getSource,
+  getSources,
+  getSourceByURL,
+  getSourceByUrlInSources
+} from "./sources";
+
+import type { Action } from "../actions/types";
+import type { SourcesMap, SourcesState } from "./sources";
+
+type Tab = string;
+export type TabList = Tab[];
+export type TabsState = Tab[];
```

Comment:
I don't know whether or not to keep both `TabList` and `TabState` or just use one.